### PR TITLE
Revamp json schema validator

### PR DESCRIPTION
**Pull request by hasuniea**
### Purpose
Revamp validation filter and wrap java methods to ballerina interop 

### Issues
https://github.com/wso2/product-microgateway/issues/1033
Fixes #

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 1]
```diff
@@ -135,9 +139,16 @@ public void execute() {
...
+            String resourcesPath =
+                    CmdUtils.getProjectTargetModulePath(projectName) + File.separator + CliConstants.RESOURCES_DIR;
+            CmdUtils.copyFolder(CmdUtils.getAPIDefinitionPath(projectName), resourcesPath);
+            // If resources folder contains .yaml file, replace the .yaml with .json file
+            replaceYAMLFilesToJson(resourcesPath);
+
             // Copy static source files
-            CmdUtils.copyAndReplaceFolder(CmdUtils.getProjectInterceptorsPath(projectName),
-                    CmdUtils.getProjectTargetInterceptorsPath(projectName));
+            //CmdUtils.copyAndReplaceFolder(CmdUtils.getProjectInterceptorsPath(projectName),
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java]: Why are we not copying this

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java [hunk 2]
```diff
@@ -146,6 +157,40 @@ public void execute() {
...
+        Object obj = yamlReader.readValue(yaml, Object.class);
+
+        ObjectMapper jsonWriter = new ObjectMapper();
+        return jsonWriter.writeValueAsString(obj);
+    }
+
+    private void replaceYAMLFilesToJson(String resPath) throws IOException {
+        String fileContent;
+        String val = null;
+        FileInputStream fileInputStream;
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java]: Streams are not closed properly

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/ext/ExtendedAPI.java [hunk 3]
```diff
@@ -33,6 +33,8 @@
     private String mgwApiScope = null;
     //isGrpc
     private boolean isGrpc = false;
+    //projectName
+    private  String projectName = null;
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/ext/ExtendedAPI.java]: Lets remove this from the API

File: tests/src/test/resources/testng.xml [hunk 4]
```diff
@@ -64,7 +64,7 @@
             <class name="org.wso2.micro.gateway.tests.extensions.OASAPIInvokeTestCase"/>
             <!--            <class name="org.wso2.micro.gateway.tests.security.CookieAuthTestCase"/>-->
             <!--            <class name="org.wso2.micro.gateway.tests.validation.ValidationTestCase"/>-->
-            <class name="org.wso2.micro.gateway.tests.interceptor.InterceptorTestCase"/>
+<!--            <class name="org.wso2.micro.gateway.tests.interceptor.InterceptorTestCase"/>-->
```
Comment (Rajith90) [inline tests/src/test/resources/testng.xml]: Whis is this commented

File: components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal [hunk 5]
```diff
@@ -40,6 +40,8 @@ map<FilterConfiguration?> filterConfigAnnotationMap = {};
 map<http:InboundAuthHandler> authHandlersMap = {}; //all handlers except for jwt handlers
 http:InboundAuthHandler[] jwtHandlers = [];//all jwt issuer handlers
 string authHeaderFromConfig = getConfigValue(AUTH_CONF_INSTANCE_ID, AUTH_HEADER_NAME, DEFAULT_AUTH_HEADER_NAME);
+map<string> openAPIs = {};
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal]: I think we can save the open APIs at the java side

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java (hunk 1 above)
Comment (hasuniea) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java]: Fixed.

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java (hunk 2 above)
Comment (hasuniea) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java]: fixed

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/ext/ExtendedAPI.java (hunk 3 above)
Comment (hasuniea) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/ext/ExtendedAPI.java]: removed

File: tests/src/test/resources/testng.xml (hunk 4 above)
Comment (hasuniea) [reply tests/src/test/resources/testng.xml]: fixed

File: components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal (hunk 5 above)
Comment (hasuniea) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/utils/utils.bal]: fixed

**Pull request merged**